roles/container-engine/docker/templates/docker-options.conf.j2

```diff
@@ -1,15 +1,17 @@
 [Service]
-Environment="DOCKER_OPTS={{ docker_options|default('') }} --iptables={{ docker_iptables_enabled | default('false') }} \
+Environment="DOCKER_OPTS=--iptables={{ docker_iptables_enabled | default('false') }} \
+{{ docker_options|default('') }} \
 {% for i in docker_insecure_registries %}--insecure-registry={{ i }} {% endfor %} \
 {% for i in docker_registry_mirrors %}--registry-mirror={{ i }} {% endfor %} \
 {% if docker_version != "latest" and docker_version is version('17.05', '<') %}--graph={% else %}--data-root={% endif %}{{ docker_daemon_graph }} \
 {% if ansible_os_family not in ["openSUSE Leap", "openSUSE Tumbleweed", "Suse"] %}{{ docker_log_opts }}{% endif %} \
+{% if ansible_system_mgr | default(false) == "systemd" %}--exec-opt native.cgroupdriver=systemd {% endif %} \
```

Comment:
your are change cgroupdriver for ALL architect 

cgroupdriver must be same for docker and kubelet !
this is broken cluster

Reply:
But here is the thing... K8s will not let you join the cluster if the cgroup driver is not systemd when you're joining a systemd based system. Thats why this issues exists in the first place.

Reply:
> K8s will not let you join the cluster if the cgroup driver is not systemd when you're joining a systemd based system.

hmmm....